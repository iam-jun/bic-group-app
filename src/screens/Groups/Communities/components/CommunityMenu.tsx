import React from 'react';
import {View, StyleSheet, ScrollView} from 'react-native';
import {useTheme} from 'react-native-paper';

import {ITheme} from '~/theme/interfaces';
import ButtonWrapper from '~/beinComponents/Button/ButtonWrapper';
import Icon from '~/beinComponents/Icon';
import Text from '~/beinComponents/Text';

const communityMenuData = [
  {
    id: 1,
    text: 'communities:community_menu:your_communities_text',
    icon: 'UsersAlt',
    type: 'COMMUNITIES',
  },
  {
    id: 2,
    text: 'communities:community_menu:manage_text',
    icon: 'Dashboard',
    type: 'MANAGE',
  },
  {
    id: 3,
    text: 'communities:community_menu:discover_text',
    icon: 'Compass',
    type: 'DISCOVER',
  },
];

const CommunityMenu = ({
  selectedIndex,
  onPress,
}: {
  selectedIndex?: number;
  onPress: (item: any, index: number) => void;
}) => {
  const theme = useTheme() as ITheme;
  const styles = createStyle(theme);

  const renderReactItem = (item: any, index: number) => {
    const isSelected = index === selectedIndex;
    return (
      <View style={styles.itemView} key={`item_community_data_${item?.text}`}>
        <ButtonWrapper
          contentStyle={[
            styles.itemContainer,
            isSelected ? styles.itemSelectedContainer : {},
          ]}
          onPress={() => {
            onPress(item, index);
          }}>
          <Icon
            icon={item.icon}
            size={24}
            tintColor={
              isSelected ? theme.colors.primary : theme.colors.iconTintLight
            }
            style={styles.icon}
          />
          <Text
            useI18n
            color={
              isSelected ? theme.colors.primary : theme.colors.textSecondary
            }>
            {item.text}
          </Text>
        </ButtonWrapper>
      </View>
    );
  };
  return (
    <View style={styles.container}>
      <ScrollView
        horizontal
        style={{backgroundColor: theme.colors.background}}
        showsHorizontalScrollIndicator={false}
        alwaysBounceHorizontal={false}>
        {communityMenuData.map(renderReactItem)}
      </ScrollView>
    </View>
  );
};

const createStyle = (theme: ITheme) => {
  const {spacing, colors} = theme;
  return StyleSheet.create({
    container: {
      paddingVertical: spacing.padding.base,
      backgroundColor: colors.background,
      borderTopWidth: 1,
      borderBottomWidth: 1,
      borderColor: colors.placeholder,
    },
    itemView: {
      padding: 0,
      margin: 0,
      marginLeft: spacing.margin.small,
    },
    itemContainer: {
      backgroundColor: colors.surface,
      flexDirection: 'row',
      padding: spacing.padding.small,
      borderRadius: 100,
    },
    itemSelectedContainer: {
      backgroundColor: colors.primary2,
    },
    iconLeftStyle: {marginRight: spacing.margin.base},
    icon: {
      marginRight: spacing.margin.small,
    },
  });
};

export default CommunityMenu;
