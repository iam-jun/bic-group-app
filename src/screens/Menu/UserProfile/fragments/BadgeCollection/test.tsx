import React from 'react';
import {
  View, Dimensions, FlatList, StyleSheet,
} from 'react-native';
import { ExtendedTheme, useTheme } from '@react-navigation/native';
import Avatar from '~/baseComponents/Avatar';
import Icon from '~/baseComponents/Icon';
import spacing, { borderRadius } from '~/theme/spacing';

const Grid = ({ data }: {data: any[]}) => {
  const deviceWidth = Dimensions.get('window').width;

  const itemWidth = 48; // Constant item width
  const numColumns = Math.floor((deviceWidth - 10 * 2) / (itemWidth + 6 * 2));
  const rowWidth = numColumns * (itemWidth + 6 * 2);
  const paddingLeft = (deviceWidth - (10 * 2) - rowWidth) / 2;

  const theme: ExtendedTheme = useTheme();
  const { colors } = theme;
  const styles = themeStyles(theme);

  const isSelected = false;

  const renderGridItem = ({ item, index }: any) =>
    // Render each grid item here
    (
      <View
        key={index}
        style={{
          margin: 6,
          borderWidth: isSelected ? 2 : 1,
          borderColor: isSelected ? colors.purple50 : colors.neutral5,
          borderRadius: borderRadius.pill,
        }}
      >
        <Avatar.Medium
          isRounded
          source={{ uri: item?.iconUrl }}
        />
        {isSelected
          ? (
            <View style={styles.iconChose}>
              <Icon size={10} icon="Check" tintColor={colors.white} />
            </View>
          )
          : null}
      </View>
    );

  return (
    <View style={{
      flex: 1,
      justifyContent: 'center',
      alignItems: 'center',
      paddingHorizontal: 10,
      paddingBottom: 10,
    }}
    >

      <View style={{
        // paddingLeft,
      }}
      >
        <FlatList
          data={data}
          renderItem={renderGridItem}
          keyExtractor={(item, index) => index.toString()}
          numColumns={numColumns}
        />
      </View>
    </View>
  );
};

const themeStyles = (theme: ExtendedTheme) => {
  const { colors } = theme;

  return StyleSheet.create({
    container: {
      flex: 1,
      paddingTop: spacing.padding.large,
    },
    iconChose: {
      backgroundColor: colors.purple50,
      borderRadius: borderRadius.pill,
      height: 20,
      width: 20,
      alignItems: 'center',
      justifyContent: 'center',
      position: 'absolute',
      zIndex: 1,
      right: 0,
    },
  });
};

export default Grid;
