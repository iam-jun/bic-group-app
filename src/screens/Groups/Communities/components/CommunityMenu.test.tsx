import * as React from 'react';
import {cleanup, render} from '@testing-library/react-native';

import CommunityMenu from './CommunityMenu';
import {renderWithRedux} from '~/test/testUtils';

afterEach(cleanup);

describe('CommunityMenu component', () => {
  it(`renders correctly`, () => {
    const onPress = jest.fn();
    const rendered = renderWithRedux(
      <CommunityMenu selectedIndex={0} onPress={onPress} />,
    ).toJSON();
    expect(rendered).toMatchSnapshot();
  });

  //   it(`renders correctly color`, () => {
  //     const {getByTestId} = render(<Divider color={'#B2BDCD'} />);
  //     const dividerComponent = getByTestId('divider');
  //     expect(dividerComponent.props.style[0].backgroundColor).toBe('#B2BDCD');
  //   });

  //   it(`renders correctly horizontal`, () => {
  //     const {getByTestId} = render(<Divider horizontal={true} />);
  //     const dividerComponent = getByTestId('divider');
  //     expect(dividerComponent.props.style[0].height).toBe(undefined);
  //   });
});
