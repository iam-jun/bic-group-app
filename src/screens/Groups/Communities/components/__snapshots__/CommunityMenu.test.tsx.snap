// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CommunityMenu component renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#FFFFFF",
      "borderBottomWidth": 1,
      "borderColor": "#EAEDF2",
      "borderTopWidth": 1,
      "paddingVertical": 12,
    }
  }
>
  <RCTScrollView
    alwaysBounceHorizontal={false}
    horizontal={true}
    showsHorizontalScrollIndicator={false}
    style={
      Object {
        "backgroundColor": "#FFFFFF",
      }
    }
  >
    <View>
      <View
        style={
          Object {
            "margin": 0,
            "marginLeft": 8,
            "padding": 0,
          }
        }
      >
        <View
          accessible={true}
          collapsable={false}
          focusable={true}
          nativeID="animatedComponent"
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
          testID="item_community_data_1"
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#EADFFE",
                "borderRadius": 100,
                "flexDirection": "row",
                "justifyContent": "center",
                "padding": 8,
              }
            }
            testID="button_wrapper.content"
          >
            <View
              accessibilityState={
                Object {
                  "disabled": true,
                }
              }
              accessible={true}
              collapsable={false}
              focusable={false}
              hitSlop={
                Object {
                  "bottom": 10,
                  "left": 10,
                  "right": 10,
                  "top": 10,
                }
              }
              nativeID="animatedComponent"
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "opacity": 1,
                }
              }
            >
              <View
                style={
                  Array [
                    Object {
                      "alignItems": "center",
                      "flexDirection": "row",
                    },
                    Object {
                      "marginRight": 8,
                    },
                    Object {
                      "backgroundColor": undefined,
                    },
                  ]
                }
              >
                <View
                  style={
                    Array [
                      undefined,
                      undefined,
                      undefined,
                    ]
                  }
                >
                  <View
                    className="icon-wrapper"
                  >
                    <RNSVGSvgView
                      align="xMidYMid"
                      bbHeight={24}
                      bbWidth={24}
                      fill="#5D2C9A"
                      focusable={false}
                      height={24}
                      meetOrSlice={0}
                      minX={0}
                      minY={0}
                      style={
                        Array [
                          Object {
                            "backgroundColor": "transparent",
                            "borderWidth": 0,
                          },
                          Object {},
                          Object {
                            "flex": 0,
                            "height": 24,
                            "width": 24,
                          },
                        ]
                      }
                      vbHeight={24}
                      vbWidth={24}
                      width={24}
                    >
                      <RNSVGGroup
                        fill={4284296346}
                        propList={
                          Array [
                            "fill",
                          ]
                        }
                      >
                        <RNSVGPath
                          d="M12.3,12.22A4.92,4.92,0,0,0,14,8.5a5,5,0,0,0-10,0,4.92,4.92,0,0,0,1.7,3.72A8,8,0,0,0,1,19.5a1,1,0,0,0,2,0,6,6,0,0,1,12,0,1,1,0,0,0,2,0A8,8,0,0,0,12.3,12.22ZM9,11.5a3,3,0,1,1,3-3A3,3,0,0,1,9,11.5Zm9.74.32A5,5,0,0,0,15,3.5a1,1,0,0,0,0,2,3,3,0,0,1,3,3,3,3,0,0,1-1.5,2.59,1,1,0,0,0-.5.84,1,1,0,0,0,.45.86l.39.26.13.07a7,7,0,0,1,4,6.38,1,1,0,0,0,2,0A9,9,0,0,0,18.74,11.82Z"
                        />
                      </RNSVGGroup>
                    </RNSVGSvgView>
                  </View>
                </View>
              </View>
            </View>
            <Text
              style={
                Object {
                  "color": "#5D2C9A",
                  "fontFamily": "OpenSans-Regular",
                  "fontSize": 16,
                }
              }
            >
              Your Communities
            </Text>
          </View>
        </View>
      </View>
      <View
        style={
          Object {
            "margin": 0,
            "marginLeft": 8,
            "padding": 0,
          }
        }
      >
        <View
          accessible={true}
          collapsable={false}
          focusable={true}
          nativeID="animatedComponent"
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
          testID="item_community_data_2"
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#F9FAFB",
                "borderRadius": 100,
                "flexDirection": "row",
                "justifyContent": "center",
                "padding": 8,
              }
            }
            testID="button_wrapper.content"
          >
            <View
              accessibilityState={
                Object {
                  "disabled": true,
                }
              }
              accessible={true}
              collapsable={false}
              focusable={false}
              hitSlop={
                Object {
                  "bottom": 10,
                  "left": 10,
                  "right": 10,
                  "top": 10,
                }
              }
              nativeID="animatedComponent"
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "opacity": 1,
                }
              }
            >
              <View
                style={
                  Array [
                    Object {
                      "alignItems": "center",
                      "flexDirection": "row",
                    },
                    Object {
                      "marginRight": 8,
                    },
                    Object {
                      "backgroundColor": undefined,
                    },
                  ]
                }
              >
                <View
                  style={
                    Array [
                      undefined,
                      undefined,
                      undefined,
                    ]
                  }
                >
                  <View
                    className="icon-wrapper"
                  >
                    <RNSVGSvgView
                      align="xMidYMid"
                      bbHeight={24}
                      bbWidth={24}
                      fill="#627286"
                      focusable={false}
                      height={24}
                      meetOrSlice={0}
                      minX={0}
                      minY={0}
                      style={
                        Array [
                          Object {
                            "backgroundColor": "transparent",
                            "borderWidth": 0,
                          },
                          Object {},
                          Object {
                            "flex": 0,
                            "height": 24,
                            "width": 24,
                          },
                        ]
                      }
                      vbHeight={24}
                      vbWidth={24}
                      width={24}
                    >
                      <RNSVGGroup
                        fill={4284641926}
                        propList={
                          Array [
                            "fill",
                          ]
                        }
                      >
                        <RNSVGPath
                          d="M19.088,4.95453c-.00732-.00781-.00952-.01819-.01715-.02582s-.01819-.00995-.02606-.01733a9.97886,9.97886,0,0,0-14.08948,0c-.00787.00738-.01837.00964-.02606.01733s-.00983.018-.01715.02582a10,10,0,1,0,14.1759,0ZM12,20a7.9847,7.9847,0,0,1-6.235-3H9.78027a2.9636,2.9636,0,0,0,4.43946,0h4.01532A7.9847,7.9847,0,0,1,12,20Zm-1-5a1,1,0,1,1,1,1A1.001,1.001,0,0,1,11,15Zm8.41022.00208L19.3999,15H15a2.99507,2.99507,0,0,0-2-2.81573V9a1,1,0,0,0-2,0v3.18427A2.99507,2.99507,0,0,0,9,15H4.6001l-.01032.00208A7.93083,7.93083,0,0,1,4.06946,13H5a1,1,0,0,0,0-2H4.06946A7.95128,7.95128,0,0,1,5.68854,7.10211l.65472.65473A.99989.99989,0,1,0,7.75732,6.34277l-.65466-.65466A7.95231,7.95231,0,0,1,11,4.06946V5a1,1,0,0,0,2,0V4.06946a7.95231,7.95231,0,0,1,3.89734,1.61865l-.65466.65466a.99989.99989,0,1,0,1.41406,1.41407l.65472-.65473A7.95128,7.95128,0,0,1,19.93054,11H19a1,1,0,0,0,0,2h.93054A7.93083,7.93083,0,0,1,19.41022,15.00208Z"
                        />
                      </RNSVGGroup>
                    </RNSVGSvgView>
                  </View>
                </View>
              </View>
            </View>
            <Text
              style={
                Object {
                  "color": "#627286",
                  "fontFamily": "OpenSans-Regular",
                  "fontSize": 16,
                }
              }
            >
              Manage
            </Text>
          </View>
        </View>
      </View>
      <View
        style={
          Object {
            "margin": 0,
            "marginLeft": 8,
            "padding": 0,
          }
        }
      >
        <View
          accessible={true}
          collapsable={false}
          focusable={true}
          nativeID="animatedComponent"
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
          testID="item_community_data_3"
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#F9FAFB",
                "borderRadius": 100,
                "flexDirection": "row",
                "justifyContent": "center",
                "padding": 8,
              }
            }
            testID="button_wrapper.content"
          >
            <View
              accessibilityState={
                Object {
                  "disabled": true,
                }
              }
              accessible={true}
              collapsable={false}
              focusable={false}
              hitSlop={
                Object {
                  "bottom": 10,
                  "left": 10,
                  "right": 10,
                  "top": 10,
                }
              }
              nativeID="animatedComponent"
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "opacity": 1,
                }
              }
            >
              <View
                style={
                  Array [
                    Object {
                      "alignItems": "center",
                      "flexDirection": "row",
                    },
                    Object {
                      "marginRight": 8,
                    },
                    Object {
                      "backgroundColor": undefined,
                    },
                  ]
                }
              >
                <View
                  style={
                    Array [
                      undefined,
                      undefined,
                      undefined,
                    ]
                  }
                >
                  <View
                    className="icon-wrapper"
                  >
                    <RNSVGSvgView
                      align="xMidYMid"
                      bbHeight={24}
                      bbWidth={24}
                      fill="#627286"
                      focusable={false}
                      height={24}
                      meetOrSlice={0}
                      minX={0}
                      minY={0}
                      style={
                        Array [
                          Object {
                            "backgroundColor": "transparent",
                            "borderWidth": 0,
                          },
                          Object {},
                          Object {
                            "flex": 0,
                            "height": 24,
                            "width": 24,
                          },
                        ]
                      }
                      vbHeight={24}
                      vbWidth={24}
                      width={24}
                    >
                      <RNSVGGroup
                        fill={4284641926}
                        propList={
                          Array [
                            "fill",
                          ]
                        }
                      >
                        <RNSVGPath
                          d="M12,2A10,10,0,1,0,22,12,10,10,0,0,0,12,2Zm1,17.93V19a1,1,0,0,0-2,0v.93A8,8,0,0,1,4.07,13H5a1,1,0,0,0,0-2H4.07A8,8,0,0,1,11,4.07V5a1,1,0,0,0,2,0V4.07A8,8,0,0,1,19.93,11H19a1,1,0,0,0,0,2h.93A8,8,0,0,1,13,19.93ZM15.14,7.55l-5,2.12a1,1,0,0,0-.52.52l-2.12,5a1,1,0,0,0,.21,1.1,1,1,0,0,0,.7.3.93.93,0,0,0,.4-.09l5-2.12a1,1,0,0,0,.52-.52l2.12-5a1,1,0,0,0-1.31-1.31Zm-2.49,5.1-2.28,1,1-2.28,2.28-1Z"
                        />
                      </RNSVGGroup>
                    </RNSVGSvgView>
                  </View>
                </View>
              </View>
            </View>
            <Text
              style={
                Object {
                  "color": "#627286",
                  "fontFamily": "OpenSans-Regular",
                  "fontSize": 16,
                }
              }
            >
              Discover
            </Text>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
